app/services/invoice_pdf_generator.rb

```diff
@@ -46,15 +46,15 @@ class InvoicePdfGenerator
   # - :client_address (String) - Address of the client
   # - :items (Array of Hashes) - List of items, each item being a hash with keys:
   #   - :item_name (String) - Name of the item
-  #   - :price_cents (Integer) - Price of the item in cents
+  #   - :price (Money) - Price of the item as a Money object
   #   - :quantity (Integer) - Quantity of the item
-  # - :payment_amount_cents (Integer, optional) - Amount already paid in cents
+  # - :payment_amount (Money, optional) - Amount already paid as a Money object
   # - :payment_date (String, optional) - Date of payment
   # - :payment_method (String, optional) - Method of payment
   def initialize(input)
     @input = input
     @doc_metadata = InvoiceLib::PDFMetadata.new(invoice_id: input[:invoice_id])
-    @total_price_in_cents = input[:items].sum { it[:price_cents] * it[:quantity] }
+    @total_price = input[:items].sum { |item| to_money(item[:price]) * item[:quantity] }
```

Comment:
FYI, Ruby has an automatic `it` parameter for blocks that take a single parameter (but you can also be explicit and pass `item` as a parameter)